loopy/schedule/checker/lexicographic_order_map.py

```diff
@@ -0,0 +1,198 @@
+        The key '0' is also include and represents a :class:`islpy.PwAff` zero
+        constant. This dictionary defines the space to be used for the set and
+        must also include versions of `dim_names` with the `in_dim_mark`
+        appended. If no value is passed, the dictionary will be made using
+        `dim_names` and `dim_names` with the `in_dim_mark` appended.
+
+    :returns: An :class:`islpy.Set` representing a big-endian lexicographic
+        ordering with the number of dimensions provided in `dim_names`. The set
+        has two dimensions for each name in `dim_names`, one identified by the
+        given name and another identified by the same name with `in_dim_mark`
+        appended. The set contains all points which meet a 'happens before'
+        constraint defining the lexicographic ordering. E.g., if
+        `dim_names = [i0, i1, i2]` and `in_dim_mark="'"`,
+        return the set containing all points in a 3-dimensional, big-endian
+        lexicographic ordering such that point
+        `[i0', i1', i2']` happens before `[i0, i1, i2]`. I.e., return::
+
+            {[i0', i1', i2', i0, i1, i2] :
+                i0' < i0 or (i0' = i0 and i1' < i1)
+                or (i0' = i0 and i1' = i1 and i2' < i2)}
+
+    """
+
+    from loopy.schedule.checker.utils import (
+        append_mark_to_strings,
+    )
+
+    in_dim_names = append_mark_to_strings(dim_names, mark=in_dim_mark)
+
+    # If no islvars passed, make them using the names provided
+    # (make sure to pass var names in desired order of space dims)
+    if islvars is None:
+        islvars = isl.make_zero_and_vars(
+            in_dim_names+dim_names,
+            [])
+
+    # Initialize set with constraint i0' < i0
+    lex_order_set = islvars[in_dim_names[0]].lt_set(islvars[dim_names[0]])
+
+    # For each dim d, starting with d=1, equality_conj_set will be constrained
+    # by d equalities, e.g., (i0' = i0 and i1' = i1 and ... i(d-1)' = i(d-1)).
+    equality_conj_set = islvars[0].eq_set(islvars[0])  # initialize to 'true'
+
+    for i in range(1, len(in_dim_names)):
+
+        # Add the next equality constraint to equality_conj_set
+        equality_conj_set = equality_conj_set & \
+            islvars[in_dim_names[i-1]].eq_set(islvars[dim_names[i-1]])
+
+        # Create a set constrained by adding a less-than constraint for this dim,
+        # e.g., (i1' < i1), to the current equality conjunction set.
+        # For each dim d, starting with d=1, this full conjunction will have
+        # d equalities and one inequality, e.g.,
+        # (i0' = i0 and i1' = i1 and ... i(d-1)' = i(d-1) and id' < id)
+        full_conj_set = islvars[in_dim_names[i]].lt_set(
+            islvars[dim_names[i]]) & equality_conj_set
+
+        # Union this new constraint with the current lex_order_set
+        lex_order_set = lex_order_set | full_conj_set
+
+    return lex_order_set
+
+
+def create_lex_order_map(
```

Comment:
Yes. I'm underscoring `_create_lex_order_set` for now, but in the future it could be useful in automatic creation of deps from legacy kernels (I use it there to create constraints, which is why it was not underscored).